File: cosmos-master/code/search/binary_search/binary_search.js
function binarySearch(array, key) {
    var lo = 0,
        hi = array.length - 1,
        mid,
        element;
    while (lo <= hi) {
        mid = Math.floor((lo + hi) / 2, 10);
        element = array[mid];
        if (element < key) {
            lo = mid + 1;
        } else if (element > key) {
            hi = mid - 1;
        } else {
            return mid;
        }
    }
    return -1;
}

File: JAVA prog/BinarySearch.java
import java.util.*;
public class BinarySearch {
    private static final Scanner sc = new Scanner(System.in);
    public static void main(String[]args) {
        int n = sc.nextInt();
        int arr[] = new int[n];
        for(int i=0; i<n; i++)
            arr[i] = sc.nextInt();
        int ele = sc.nextInt();
        int res = BinarySearch(arr,ele);
        if(res == -1)
            System.out.println("Not Found");
        else
            System.out.println("Fount at = "+res);
    }
    static int BinarySearch(int [] arr, int ele) {
        int l = 0;
        int r = arr.length -1;
        while(l <= r) {
            int mid = l+(r-l)/2;
            if(arr[mid] == ele)
                return mid;
            if(arr[mid] < ele)
                l = mid + 1;
            else
                r = mid - 1;
        }
        return -1;
    }
}
